projects/LiteNetLibTest/Media/Output/OggAudioPlayer.cs

```diff
@@ -0,0 +1,131 @@
+    private readonly IWavePlayer player;
+
+    /// <inheritdoc />
+    public int StreamSerialNo { get; }
+
+    /// <inheritdoc />
+    public byte[] StreamHeader { get; }
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    /// <param name="streamSerialNumber">Stream serial number.</param>
+    /// <param name="sampleRate">Sample rate (hertz).</param>
+    public OggAudioPlayer(int streamSerialNumber, int sampleRate)
+    {
+        StreamSerialNo = streamSerialNumber;
+
+        // Initialize player.
+        bufferedWaveProvider = new BufferedWaveProvider(
+            waveFormat: WaveFormat.CreateIeeeFloatWaveFormat(sampleRate, channels: 1))
+        {
+            BufferDuration = TimeSpan.FromSeconds(400),
+        };
+
+        player = new WasapiOut();
+        player.Init(bufferedWaveProvider);
+
+        var memoryStream = new MemoryStream();
+        var oggHeaderStream = PcmAudioStreamReader.InitializeStream(streamSerialNumber, sampleRate);
+        OggDataInput.FlushPages(memoryStream, oggHeaderStream);
+        StreamHeader = memoryStream.ToArray();
+    }
+
+    /// <inheritdoc />
+    public bool Enqueue(byte[] buffer, Memory<byte>[] _)
+    {
+        if (buffer.Any())
+        {
+            // Copy to stream.
+            using var stream = new MemoryStream(StreamHeader.Concat(buffer).ToArray());
+            using var vorbisStream = new VorbisWaveReader(stream, closeOnDispose: true);
+            vorbisStream.NextStreamIndex = StreamSerialNo;
+
+            try
+            {
+                var num = 0;
+                do
+                {
+                    // TODO: not very reliable. Audio is stuttering.
+                    // For some reason, output (.raw) file is 2 times larger than the source.
+                    // Does it count header data?
+                    num = vorbisStream.Read(this.buffer, 0, 8820);
+                    file.Write(this.buffer, 0, num);
+
+                    bufferedWaveProvider.AddSamples(this.buffer, 0, num);
+                }
+                while (num > 0);
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex);
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Start worker thread to play buffered audio.
+    /// </summary>
+    public async void PlayAsync()
+    {
+        while (true)
+        {
+            await Task.Delay(millisecondsDelay: 2_000);
+
+            var completion = new TaskCompletionSource();
+            void OnPlaybackStopped(object? sender, StoppedEventArgs e)
+            {
+                if (e.Exception != null)
+                {
+                    Debug.WriteLine(e.Exception.Message);
+                }
+
+                player.PlaybackStopped -= OnPlaybackStopped;
+                completion.TrySetResult();
+            }
+
+            player.PlaybackStopped += OnPlaybackStopped;
+            player.Play();
```

Comment:
According to doc this is not a blocking call: https://github.com/naudio/NAudio/wiki/Playing-an-Audio-File#starting-playback . Why do we need to make it in a separate task?